### PR TITLE
Load dtb data and store in vm struct for later

### PR DESCRIPTION
**Pull request by NunoDasNeves**
Signed-off-by: Nuno Das Neves <nudasnev@microsoft.com>

### CODE REVIEW

**Comment by NunoDasNeves** [comment]
clippy error is from upstream

**Review by russell-islam — APPROVED** [review]

File: vmm/src/vm.rs [hunk 1]
```diff
@@ -505,6 +510,8 @@ pub struct Vm {
     stop_on_boot: bool,
     #[cfg(target_arch = "x86_64")]
     load_payload_handle: Option<thread::JoinHandle<Result<EntryPoint>>>,
+    #[cfg(target_arch = "aarch64")]
+    dtb_data: Option<Vec<u8>>,
```
Comment (russell-islam) [inline vmm/src/vm.rs:514]: I think it's not wise to keep the data in memory while we don't need it most of the time. We can keep the file and read it on demand.

Comment (NunoDasNeves) [reply vmm/src/vm.rs:514]: Is there a specific reason you think it's better as a File? We have to get it in vector form before we can parse it.

Comment (russell-islam) [reply vmm/src/vm.rs:514]: The reason is VM struct is an in-memory data structure, that stays within the VM lifetime. Would we keep the data in memory for the whole time? If it is not too much data I think it's fine. That is up to you.

**Pull request closed without merging**